[feat:#1001]: support promql

joyant opened the pull request:
### What problem does this PR solve?

Issue Number: #1001

Problem Summary:

"Benefiting from Prometheus being developed in Golang, we can import certain packages from Prometheus. This can be achieved by implementing the Querier interface to execute PromQL queries. I am not sure if everyone considers this to be a good approach.

**Currently, I have only implemented the read apis and remote write api, just enough for Grafana usage.** The apis related to setting alerts may need to be implemented after LinDB has its own alerting functionality.

Points to note:
1. Prometheus histograms and LinDB histograms are not very similar in implementation, so I did not use LinDB histograms to implement Prometheus features.
2. I believe counter, gauge, histogram, and summary should correspond to LinDB's Last Field. This needs discussion for confirmation or negation.
3. A portion of helper functions and error handling has been copied from Prometheus(Apache License).

Running and testing:
1. Append the following configuration to broker.toml or standalone.toml:
   ```
   [prometheus]
   namespace = "default-ns"
   database = "prometheus"
   field = "x"
   ```
2. When adding a data source in the Grafana admin panel, for example, in the local, it should be written as:
   ```
   localhost:9000/prometheus
   ```
4. During the development of this feature, I discovered a bug in the trie, so it is necessary to review #1012 first."

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] No code

stone1100 commented:
Can you fix some golangci and UT failed cases?

**make test** can check golangci and UT.

joyant commented:
Hi @stone1100 

I've made the necessary revisions and the latest commit now passes the automated tests, including compliance with the golangci-lint checks. Could you please take another look at the changes?

Here's a brief summary of what was done:

1. Refactored code to adhere to golangci-lint standards.
2. Addressed all linting issues that were previously highlighted.
3. Ensured that all unit and integration tests are passing.

I believe the code is now in good shape for merge consideration. Your further feedback would be greatly appreciated.

Thank you for your time and assistance!

Best regards,
joyant.
